tests/e2e/test_actions.py

```diff
@@ -384,3 +384,36 @@ def test_shutdown_action_now(update_environment):
         assert (
             "This condition should not fire" not in result.stdout
         ), "a post-shutdown condition fired when it should not have"
+
+
+@pytest.mark.e2e
+def test_inventory_as_dir():
```

Comment:
@Alex-Izquierdo Can we check that this test is broken without the changes to action/run_playbook.py and util.py.

Reply:
```
(.venv) ➜  alex@alextop  ~/repos/redhat/ansible/eda/ansible-rulebook git:(main) ✗ git checkout -b "test-inventory-dir"
Switched to a new branch 'test-inventory-dir'
(.venv) ➜  alex@alextop  ~/repos/redhat/ansible/eda/ansible-rulebook git:(test-inventory-dir) ✗ git cherry-pick 33b1ff8c09a8a67cc0f5fff569dfc25f4a044b06
[test-inventory-dir 713f8bc] test: add e2e tests for a inventory passed as dir
 Date: Thu May 16 18:29:25 2024 +0200
 6 files changed, 61 insertions(+)
 create mode 100644 tests/e2e/files/inventories/inventory_as_dir/group_vars/customgroup.yml
 create mode 100644 tests/e2e/files/inventories/inventory_as_dir/host_vars/localhost.yml
 create mode 100644 tests/e2e/files/inventories/inventory_as_dir/hosts.yml
 create mode 100644 tests/e2e/files/playbooks/print_group_vars.yml
 create mode 100644 tests/e2e/files/rulebooks/test_inventory_as_dir.yml
(.venv) ➜  alex@alextop  ~/repos/redhat/ansible/eda/ansible-rulebook git:(test-inventory-dir) ✗ pytest -m e2e -k test_inventory_as_dir
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.11.6, pytest-7.4.4, pluggy-1.3.0
rootdir: /home/alex/repos/redhat/ansible/eda/ansible-rulebook
configfile: pytest.ini
plugins: timeout-2.2.0, xdist-3.5.0, cov-4.1.0, check-2.3.0, asyncio-0.23.3, jira-0.3.18
asyncio: mode=Mode.STRICT
collected 309 items / 308 deselected / 1 selected                                                                                                                                                                 

tests/e2e/test_actions.py::test_inventory_as_dir 
-------------------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------------------
2024-05-16 19:22:11 INFO Running command: ansible-rulebook -i /home/alex/repos/redhat/ansible/eda/ansible-rulebook/tests/e2e/files/inventories/inventory_as_dir -S /home/alex/repos/redhat/ansible/eda/ansible-rulebook/tests/e2e/../sources --rulebook /home/alex/repos/redhat/ansible/eda/ansible-rulebook/tests/e2e/files/rulebooks/test_inventory_as_dir.yml
FAILED                                                                                                                                                                                                      [100%]

==================================================================================================== FAILURES =====================================================================================================
______________________________________________________________________________________________ test_inventory_as_dir ______________________________________________________________________________________________

    @pytest.mark.e2e
    def test_inventory_as_dir():
        """
        Execute a rulebook that contains a run_playbook action with an inventory
        directory instead of a file.
        """
    
        rulebook = utils.BASE_DATA_PATH / "rulebooks/test_inventory_as_dir.yml"
        inventory = utils.BASE_DATA_PATH / "inventories/inventory_as_dir"
        cmd = utils.Command(
            rulebook=rulebook,
            inventory=inventory,
        )
    
        LOGGER.info(f"Running command: {cmd}")
        result = subprocess.run(
            cmd,
            timeout=DEFAULT_CMD_TIMEOUT,
            capture_output=True,
            cwd=utils.BASE_DATA_PATH,
            text=True,
        )
    
        assert result.returncode == 0
        assert not result.stderr
>       assert (
            "hostvar_value" in result.stdout
        ), "hostvar_value not found in stdout"
E       AssertionError: hostvar_value not found in stdout
E       assert 'hostvar_value' in '\x1b[1;35m[WARNING]: Could not match supplied host pattern, ignoring: customgroup\x1b[0m\n\nPLAY [customgroup] *************************************************************\n\x1b[0;36mskipping: no hosts matched\x1b[0m\n\nPLAY RECAP *********************************************************************\n'
E        +  where '\x1b[1;35m[WARNING]: Could not match supplied host pattern, ignoring: customgroup\x1b[0m\n\nPLAY [customgroup] *************************************************************\n\x1b[0;36mskipping: no hosts matched\x1b[0m\n\nPLAY RECAP *********************************************************************\n' = CompletedProcess(args=Command(rulebook=PosixPath('/home/alex/repos/redhat/ansible/eda/ansible-rulebook/tests/e2e/files/rulebooks/test_inventory_as_dir.yml'), program_name='ansible-rulebook', cwd=PosixPath('/home/alex/repos/redhat/ansible/eda/ansible-rulebook/tests/e2e/files'), inventory=PosixPath('/home/alex/repos/redhat/ansible/eda/ansible-rulebook/tests/e2e/files/inventories/inventory_as_dir'), sources=PosixPath('/home/alex/repos/redhat/ansible/eda/ansible-rulebook/tests/e2e/../sources'), vars_file=None, envvars=None, proc_id=None, verbose=False, debug=False, websocket=None, token_url=None, access_token=None, refresh_token=None, project_tarball=None, worker_mode=False, verbosity=0, heartbeat=0, execution_strategy=None, hot_reload=False, skip_audit_events=False, vault_ids=None, vault_password_file=None), returncode=0, stdout='\x1b[1;35m[WARNING]: Could not match supplied host pattern, ignoring: customgroup\x1b[0m\n\nPLAY [customgroup] *************************************************************\n\x1b[0;36mskipping: no hosts matched\x1b[0m\n\nPLAY RECAP *********************************************************************\n', stderr='').stdout

tests/e2e/test_actions.py:414: AssertionError
------------------------------------------------------------------------------------------------ Captured log call ------------------------------------------------------------------------------------------------
2024-05-16 19:22:11 INFO Running command: ansible-rulebook -i /home/alex/repos/redhat/ansible/eda/ansible-rulebook/tests/e2e/files/inventories/inventory_as_dir -S /home/alex/repos/redhat/ansible/eda/ansible-rulebook/tests/e2e/../sources --rulebook /home/alex/repos/redhat/ansible/eda/ansible-rulebook/tests/e2e/files/rulebooks/test_inventory_as_dir.yml
============================================================================================= short test summary info =============================================================================================
FAILED tests/e2e/test_actions.py::test_inventory_as_dir - AssertionError: hostvar_value not found in stdout
======================================================================================== 1 failed, 308 deselected in 4.25s ========================================================================================

```

Reply:
@Alex-Izquierdo My mistake I was looking for the error reported in the ticket which point to a file instead of a directory